python/api/onnxruntime-python-api.py

```diff
@@ -0,0 +1,127 @@
+# A set of code samples showing different usage of the ONNX Runtime Python API
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import torch
+import onnxruntime
+
+MODEL_FILE = '.model.onnx'
+DEVICE_NAME = 'cuda' if torch.cuda.is_available() else 'cpu'
+DEVICE_INDEX = 0     # Replace this with the index of the device you want to run on
+DEVICE=f'{DEVICE_NAME}:{DEVICE_INDEX}'
+
+# A simple model to calculate addition of two tensors
+def model():
+    class Model(torch.nn.Module):
+        def __init__(self):
+            super(Model, self).__init__()
+
+        def forward(self, x, y):
+            return x.add(y)
+
+    return Model()
+
+# Create an instance of the model and export it to ONNX graph format, with dynamic size for the data
+def create_model(type: torch.dtype = torch.float32):
+    sample_x = torch.ones(3, dtype=type)
+    sample_y = torch.zeros(3, dtype=type)
+
+    torch.onnx.export(model(), (sample_x, sample_y), MODEL_FILE, input_names=["x", "y"], output_names=["z"],
+                               dynamic_axes={"x": {0 : "array_length_x"}, "y": {0: "array_length_y"}})
+ 
+# Create an ONNX Runtime session with the provided model
+def create_session(model: str) -> onnxruntime.InferenceSession:
+    providers = ['CPUExecutionProvider']
+    if torch.cuda.is_available():
+        providers.insert(0, 'CUDAExecutionProvider')
+    return onnxruntime.InferenceSession(model, providers=providers)
+
+# Run the model on CPU consuming and producing numpy arrays 
+def run(x: np.array, y: np.array) -> np.array:
+    session = create_session(MODEL_FILE)
+
+    z = session.run(["z"], {"x": x, "y": y})
+    
+    return z[0]   
+
+# Run the model on device consuming and producing ORTValues
+def run_with_data_copied_to_device(x: np.array, y: np.array) -> onnxruntime.OrtValue:
```

Comment:
Nit: Can this be `run_with_iobinding` ? Or is the idea to be explicit about what is being done ? One tiny pedantic inaccuracy with this naming is that when this model runs on CPU (when CUDA is not available), then no "data copy to device" occurs, does it ?

Reply:
Would it be better named run_with_data_on_device()? (Then it is copied if GPU and already on device if CPU)